controller/array_action/array_mediator_xiv.py

```diff
@@ -305,26 +302,29 @@ def get_object_by_id(self, object_id, object_type):
             return self._generate_snapshot_response(cli_object)
         return self._generate_volume_response(cli_object)
 
-    def create_snapshot(self, name, volume_name, pool_id=None):
-        logger.info("creating snapshot {0} from volume {1}".format(name, volume_name))
-
+    def create_snapshot(self, volume_id, snapshot_name, pool=None):
+        logger.info("creating snapshot {0} from volume {1}".format(snapshot_name, volume_id))
+        cli_source_volume = self._get_cli_object_by_wwn(volume_id)
```

Comment:
shouldn't it be `source_cli_volume` (similar to `source_api_volume`)?

Reply:
done